src/AMSlib/ml/surrogate.hpp

```diff
@@ -393,6 +393,23 @@ class SurrogateModel
   }
 
   bool is_DeltaUQ() { return _is_DeltaUQ; }
+
+  void update(std::string new_path)
+  {
+    /* This function updates the underlying torch model,
+     * with a new one pointed at location modelPath. The previous
+     * one is destructed automatically.
+     *
+     * TODO: I decided to not update the model path on the ``instances''
```

Comment:
I think that's okay for now. However, we should definitely track internally at least for our own tests.

Reply:
I agree, I wasn't sure if this needs to be tracked by the DB, the WF or some other entity that tracks the provenance.